soc/arm/microchip_mec/common/spigen/mec_spi_gen.py

```diff
@@ -0,0 +1,476 @@
+    spi_drive_str_mult = args.spi_drive_strength
+    spi_slew = args.spi_slew_rate
+
+    spi_size = args.spi_size_kb * 1024
+
+    indata = None
+    with open(args.infilename, "rb") as fin:
+        indata = fin.read()
+
+    indata_len = len(indata)
+    if verbose_mode:
+        print("Read input FW binary: length = {0}".format(indata_len))
+
+    # if necessary pad input data to PLD_GRANULARITY required by Boot-ROM loader
+    if (indata_len % PLD_GRANULARITY) != 0:
+        pad_len = PLD_GRANULARITY - (indata_len % PLD_GRANULARITY)
+        # NOTE: MCHP Production SPI Image Gen. pads with 0
+        padding = PLD_PAD_BYTE * pad_len
+        indata = indata + padding
+
+    if verbose_mode:
+        print("Padded FW binary: length = {0}".format(len(indata)))
+
+    indata_len = len(indata)
+
+    # Do we have enough space for 4KB block containing TAG and Header, padded FW binary,
+    # EC Info Block, Co-Sig Block, and Trailer?
+    mec_add_info_size = PLD_SPI_LOC + EC_INFO_BLOCK_SIZE + COSIG_SIZE + TRAILER_SIZE
+    if indata_len > (spi_size - mec_add_info_size):
+        print("ERROR: FW binary exceeds flash size!")
+        sys.exit(-1)
+
+    entry_point = args.entry_point
+    if args.entry_point == 0:
+        # Look up entry point in image
+        # Assumes Cortex-M4 vector table
+        # at beginning of image and second
+        # word in table is address of reset handler
+        entry_point = int.from_bytes(indata[4:8], byteorder="little")
+
+    tag = build_tag(HDR_SPI_LOC)
+
+    if verbose_mode:
+        print_bytes("TAG", tag)
+        print("Build Header at {0}: Load Address = 0x{1:0x} Entry Point Address = 0x{2:0x}".format(HDR_SPI_LOC, PLD_LOAD_ADDR, entry_point))
+
+    spi_dict = { "spi_freq_mhz": spi_freq_mhz, "spi_mode": spi_mode,
+                 "spi_read_mode": spi_read_mode, "spi_drive_str": spi_drive_str_mult,
+                 "spi_slew_rate": spi_slew }
+
+    header = build_header(chip, spi_dict, HDR_SPI_LOC, PLD_SPI_LOC, entry_point, indata_len)
+
+    if verbose_mode:
+        print_bytes("HEADER", header)
+        print("")
+
+    # appended to end of padded payload
+    ec_info_block = bytearray(EC_INFO_BLOCK_SIZE)
+    ec_info_loc = PLD_SPI_LOC + len(indata)
+
+    # appended to end of (padded payload + ec_info_block)
+    cosig = bytearray(b'\xff' * COSIG_SIZE)
+    cosig_loc = ec_info_loc + EC_INFO_BLOCK_SIZE
+
+    # appended to end of (padded payload + ec_info_block + cosig)
+    # trailer[0:0x30] = SHA384(indata || ec_info_block || cosig)
+    # trailer[0x30:] = 0xFF
+    trailer = bytearray(b'\xff' * TRAILER_SIZE)
+    trailer_loc = cosig_loc + COSIG_SIZE
+
+    h = hashlib.sha384()
+    h.update(indata)
+    h.update(ec_info_block)
+    h.update(cosig)
+    image_digest = h.digest()
+    trailer[0:len(image_digest)] = image_digest
+
+    if verbose_mode:
+        print("SHA-384 digest (paddedFW || ec_info_block || cosig)")
+        print_bytes("digest", image_digest)
+
+    spi_bufs = []
```

Comment:
Would it make more sense to use a dictionary here? Then you could iterate it over it with just sorted(spi_bufs) and unpack the values directly using items(), which I think would make the code below a bit easier to understand.

Reply:
prefer to use list.